packages/appcd-fs/src/fs.js

```diff
@@ -86,3 +86,28 @@ export function locate(dir, filename, depth) {
 	}
 	return null;
 }
+
+/**
+ * Read a directory including scoped packages as a single entry in the Array.
+ *
+ * @param {String} dir - Directory to read.
+ * @returns {Array}
+ */
+export function readdirScoped(dir) {
+	let children = fs.readdirSync(dir);
+	const scopes = children.filter(dirName => dirName.charAt(0) === '@');
+	// Remove all scoped directories from the child array
+	children = children.filter(dirName => dirName.charAt(0) !== '@');
+	if (scopes.length === 0) {
+		return children;
+	}
+	for (const scope of scopes) {
+		const scopedDir = path.join(dir, scope);
+		const scopedChildren = fs.readdirSync(scopedDir)
+			.filter(child => isDir(child))
+			.map(child => `${scope}/${child}`);
+
+		children.push(...scopedChildren);
+	}
+	return children.sort((a, b) => a > b);
```

Comment:
1. I added sorting to make sure it's as similar to the output of a readdir as possible because we were adding to the array which put things in a different order. With the changes above I guess it wouldn't be necessary as the order would be as it comes from readdir. 

2. Yeah but we were then screwing up that order

3&4. It's just something I picked up because of the whole .sort being unstable thing, I'm probably just being nooby :), if I go for similar to your code with the one for loop this wont be needed

I'll update to have a similar flow to yours but I think i'll keep the charAt check instead of a regex, I'm posit that there's a significant performance difference there and I love a micro-optimization :)

Reply:
Yeah, `charAt()` is pretty snappy, but you know what's faster?

https://jsperf.com/charat-vs-regex-vs-startswith-vs-by-index

:)